### PR TITLE
Add named export

### PR DESCRIPTION
**Pull request by DaSchTour**
To avoid issues with default exports like
`apollo_server_plugin_response_cache_1.default is not a function`

### CODE REVIEW

**Comment by glasser** [comment]
Is there a general problem where default exports don't work for you? It's a common pattern in npm packages, and we use it in a few other places in this repo, so just making this one change doesn't make too much sense. How are you writing your import/require such that it doesn't work?

**Comment by DaSchTour** [comment]
@glasser I repeatedly had problems with default exports in different projects. I see this as a anti-pattern in JS development. In this case I had the issue when using the `apollo-server-plugin-response-cache` in a nest.js project within an nx workspace. I finally solved it by using `import * as responseCachePlugin from 'apollo-server-plugin-response-cache';` instead of `import responseCachePlugin from 'apollo-server-plugin-response-cache';` and adding it like this `plugins: [responseCachePlugin as any],`into the plugin. Although I can't really verify if it works. I don't see that the cache is really used, but I don't know if it's because of the import or something else.

**Comment by DaSchTour** [comment]
Funny thing is, that `plugins: [(responseCachePlugin as any)()],` also works (doesn't throw any error). But I don't see any requests beeing cached.

File: packages/apollo-server-plugin-response-cache/src/index.ts [hunk 1]
```diff
@@ -1,3 +1,4 @@
 import plugin from './ApolloServerPluginResponseCache';
 export default plugin;
+export { plugin as ApolloServerPluginResponseCache };
 module.exports = plugin;
```
Comment (glasser) [inline packages/apollo-server-plugin-response-cache/src/index.ts:4]: Actually, there is something weird here. Why does this file have both `export default` and `module.exports =`? That's the problem, which I'll fix.

**Comment by glasser** [comment]
I suspect that https://github.com/apollographql/apollo-server/pull/5542 fixes the issue. While I agree that named exports are generally nicer than defaults, I think "you can use either default or a name to get the same thing" is worse than either choice so for these particular packages I'm going to leave them as the defaults.

As far as getting caching working goes, are you seeing cache-control response headers set? A post in GH discussions or community.apollographql.com might be a good place to get help with that too.

**Comment by DaSchTour** [comment]
I'm not sure if you saw, that it's possible to do very strange things with this default export and it only works with typecasting, because the typings do not match the import. That's very bad developer experience. And this is also one of the few apollo packages that have a default export.

**Comment by glasser** [comment]
@DaSchTour If the "very strange things" still occur in `apollo-server-plugin-response-cache` 3.1.0 or newer, let me know. Otherwise I'll assume it's just an artifact of the weird line removed in #5542.

**Comment by DaSchTour** [comment]
@glasser following the documentation using 3.1.1 I get this error

```
 TS2322: Type '(configService: ConfigService) => { debug: boolean; tracing: boolean; playground: boolean; introspection: boolean; autoSchemaFile: string; cacheControl: { defaultMaxAge: number; }; cache: RedisCache | InMemoryLRUCache<...>; plugins: ApolloServerPlugin<...>[]; context: ({ req, res }: any) => { ...; }; }' is not assignable to type '(...args: any[]) => GqlModuleOptions | Promise<GqlModuleOptions>'.
  Type '{ debug: boolean; tracing: boolean; playground: boolean; introspection: boolean; autoSchemaFile: string; cacheControl: { defaultMaxAge: number; }; cache: RedisCache | InMemoryLRUCache<string>; plugins: ApolloServerPlugin<...>[]; context: ({ req, res }: any) => { ...; }; }' is not assignable to type 'GqlModuleOptions | Promise<GqlModuleOptions>'.
    Type '{ debug: boolean; tracing: boolean; playground: boolean; introspection: boolean; autoSchemaFile: string; cacheControl: { defaultMaxAge: number; }; cache: RedisCache | InMemoryLRUCache<string>; plugins: ApolloServerPlugin<...>[]; context: ({ req, res }: any) => { ...; }; }' is not assignable to type 'GqlModuleOptions'.
      Types of property 'plugins' are incompatible.
        Type 'ApolloServerPlugin<BaseContext>[]' is not assignable to type 'PluginDefinition[]'.
          Type 'ApolloServerPlugin<BaseContext>' is not assignable to type 'PluginDefinition'.
            Type 'import("/…/node_modules/apollo-server-plugin-response-cache/node_modules/apollo-server-plugin-base/dist/index").ApolloServerPlugin<import("/…/node_modules/apollo-server-plugin-response-cache/node_modules/apollo-server-types/dist/index").BaseContext>' is not assignable to type 'import("/…/node_modules/apollo-server-plugin-base/dist/index").ApolloServerPlugin<import("/…/node_modules/apollo-server-types/dist/index").BaseContext>'.
              Types of property 'requestDidStart' are incompatible.
                Type '(requestContext: GraphQLRequestContext<BaseContext>) => Promise<void | GraphQLRequestListener<BaseContext>>' is not assignable to type '(requestContext: GraphQLRequestContext<BaseContext>) => void | GraphQLRequestListener<BaseContext>'.
                  Types of parameters 'requestContext' and 'requestContext' are incompatible.
                    Type 'import("/…/node_modules/apollo-server-types/dist/index").GraphQLRequestContext<import("/…/node_modules/apollo-server-types/dist/index").BaseContext>' is not assignable to type 'import("/…/node_modules/apollo-server-plugin-response-cache/node_modules/apollo-server-types/dist/index").GraphQLRequestContext<import("/…/node_modules/apollo-server-plugin-response-cache/node_modules/apollo-server-types/dist/index").BaseContext>'.
                      The types of 'request.http.headers' are incompatible between these types.
                        Type 'import("/…/node_modules/apollo-server-env/dist/fetch").Headers' is not assignable to type 'import("/…/node_modules/apollo-server-plugin-response-cache/node_modules/apollo-server-env/dist/fetch").Headers'.

```

**Comment by glasser** [comment]
This isn't about named exports. You have two copies of `apollo-server-env` installed (one nested under `apollo-server-plugin-response-cache`). We had to make an incompatible change to that package in v3.1.1 as a bug fix (possibly doing this in a patch release was bad, but it was for compatibility with changes that `@types/node-fetch` made in a patch release). Can you make sure all of your packages are at the latest version (or share your package-lock.json) and perhaps run `npm dedupe`?

**Pull request closed without merging**